packages/react-import-remote/src/ImportRemote.tsx

```diff
@@ -1,73 +1,38 @@
 import * as React from 'react';
 import {Preconnect} from '@shopify/react-html';
-import {
-  RequestIdleCallbackHandle,
-  WindowWithRequestIdleCallback,
-  DeferTiming,
-} from '@shopify/async';
-import load from './load';
+import {DeferTiming} from '@shopify/async';
+import useImportRemote from './hooks';
 
 export interface Props<Imported = any> {
   source: string;
   nonce?: string;
   preconnect?: boolean;
   onError(error: Error): void;
-  getImport(window: Window): Imported;
+  getImport?(window: Window): Imported;
   onImported(imported: Imported): void;
   defer?: DeferTiming;
 }
 
-export default class ImportRemote extends React.PureComponent<Props, never> {
-  private idleCallbackHandle?: RequestIdleCallbackHandle;
+export default function ImportRemote(props: Props) {
+  const {source, preconnect, onError, onImported, nonce, getImport} = props;
+  const [loading, global, error] = useImportRemote(source, {nonce, getImport});
 
-  componentWillUnmount() {
-    if (this.idleCallbackHandle != null && 'cancelIdleCallback' in window) {
-      (window as WindowWithRequestIdleCallback).cancelIdleCallback(
-        this.idleCallbackHandle,
-      );
-    }
+  if (preconnect) {
+    const url = new URL(source);
+    return <Preconnect source={url.origin} />;
   }
 
-  async componentDidMount() {
-    if (
-      this.props.defer === DeferTiming.Idle &&
-      'requestIdleCallback' in window
-    ) {
-      this.idleCallbackHandle = (window as WindowWithRequestIdleCallback).requestIdleCallback(
-        this.loadRemote,
-      );
-    } else {
-      await this.loadRemote();
-    }
+  if (loading) {
+    return null;
   }
 
-  async componentDidUpdate({source: oldSource}: Props) {
-    const {source} = this.props;
-
-    if (oldSource !== source) {
-      await this.loadRemote();
-    }
+  if (error && onError) {
+    onError(error);
   }
 
-  render() {
-    const {source, preconnect} = this.props;
-
-    if (preconnect) {
-      const url = new URL(source);
-      return <Preconnect source={url.origin} />;
-    }
-
-    return null;
+  if (global && onImported) {
+    onImported(global);
```

Comment:
We shouldn't be calling these callbacks in render itself, I think these want to be more like:
```
useEffect(() => {
    if (global) {
      onImported(global)
    }
}, [global])
```
```
useEffect(() => {
   if (error) {
      onError(error)
    }
}, [error])
```